### PR TITLE
Option to change comments-server (desktop)

### LINKED ISSUE #5459: Interact directly with commenting servers
- [x] Drop SDK calls (SDK is dropping `comment_xxx` commands)
- [x] Replace with direct calls to server matching `commentron` APIs
- [ ] Add setting for changing your comment server. Visible on desktop (and possibly defaulting to Odysee URL), hidden on odysee.
- [ ] Revise comment onboarding on desktop.

### PR DESCRIPTION
**Pull request by infinite-persistence**
## Issue
Closes #5459
> _Add setting for changing your comment server. Visible on desktop (and possibly defaulting to Odysee URL), hidden on odysee._

## Changes
- Added Desktop-only options to define custom server. [Settings > Advanced Settings > "Comment server" section].

## Comments
Not sure how this would actually work properly without the user recompiling the app to handle server differences. For example, even when we use our own server but switch between v1 and v2, some code changes are need to handle the differences. At that point, it seems easier for the user to just change the .env file? 

But I guess it is still useful as a quick way to switch servers when there are no API interface issues.

### CODE REVIEW

**Comment by infinite-persistence** [comment]
fabbcee to dce101c
- Rebased to `master` to resolve string conflicts

File: ui/comments.js [hunk 1]
```diff
@@ -31,8 +38,10 @@ const Comments = {
 };
 
 function fetchCommentsApi(method: string, params: {}) {
-  if (!Comments.enabled) {
-    return Promise.reject('Comments are not currently enabled'); // eslint-disable-line
+  if (!Comments.url) {
+    return Promise.reject(new Error('Comments-server URL not defined.'));
```
Comment (kauffj) [inline ui/comments.js:42]: Commenting server is not set.

File: ui/component/settingCommentsServer/view.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return () => clearTimeout(timer);
+  }, [customUrl, setCustomServerUrl]);
+
+  return (
+    <React.Fragment>
+      <fieldset-section>
+        <FormField
+          type="radio"
+          name="use_default_comments_server"
+          label={__('Use official Odysee comments server')}
```
Comment (kauffj) [inline ui/component/settingCommentsServer/view.jsx:34]: `__('Default comments server (%DEFAULT_COMMENT_SERVER_NAME%)', { DEFAULT_COMMENT_SERVER_NAME })`

with the default server name moved to a config

Comment (kauffj) [reply ui/component/settingCommentsServer/view.jsx:34]: alternatively, add an array config value for all known servers, and make this a select with custom as the last option

File: ui/component/settingCommentsServer/view.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          onChange={(e) => {
+            if (e.target.checked) {
+              setCustomServerEnabled(false);
+            }
+          }}
+        />
+        <FormField
+          type="radio"
+          name="use_custom_comments_server"
+          label={__('Use custom comments server')}
```
Comment (kauffj) [inline ui/component/settingCommentsServer/view.jsx:45]: Custom comments server

(drop use)

File: ui/redux/actions/comments.js [hunk 4]
```diff
@@ -96,20 +98,32 @@ export function doCommentList(
...
-            },
-          });
-        } else {
-          devToast(dispatch, `doCommentList: ${error.message}`);
-          dispatch({
-            type: ACTIONS.COMMENT_LIST_FAILED,
-            data: error,
-          });
+        switch (error.message) {
+          case 'comments are disabled by the creator':
```
Comment (kauffj) [inline ui/redux/actions/comments.js:102]: where is this coming from? can/should it be a const?

**Review by kauffj — CHANGES_REQUESTED** [review]
I do not need to review this after these changes, just ship it.

File: ui/redux/actions/comments.js (hunk 4 above)
Comment (infinite-persistence) [reply ui/redux/actions/comments.js:102]: :+1: Yup, recently cleaned up to be constants in `master`.

**Pull request merged**